src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidRunCommand.cs

```diff
@@ -0,0 +1,261 @@
+                            instrumentationExitCode = (int)ExitCode.RETURN_CODE_NOT_SET;
+                        }
+                    }
+
+                    // Optionally copy off an entire folder
+                    if (!string.IsNullOrEmpty(deviceOutputFolder))
+                    {
+                        try
+                        {
+                            var logs = runner.PullFiles(deviceOutputFolder, outputDirectory);
+                            foreach (string log in logs)
+                            {
+                                logger.LogDebug($"Found output file: {log}");
+                            }
+                        }
+                        catch (Exception toLog)
+                        {
+                            logger.LogError(toLog, "Hit error (typically permissions) trying to pull {filePathOnDevice}", deviceOutputFolder);
+                            failurePullingFiles = true;
+                        }
+                    }
+
+                    runner.DumpAdbLog(Path.Combine(outputDirectory, $"adb-logcat-{apkPackageName}.log"));
+
+                    if (processCrashed)
+                    {
+                        runner.DumpBugReport(Path.Combine(outputDirectory, $"adb-bugreport-{apkPackageName}.zip"));
+                    }
+                }
+
+                if (instrumentationExitCode != expectedExitCode)
+                {
+                    logger.LogError($"Non-success instrumentation exit code: {instrumentationExitCode}, expected: {expectedExitCode}");
+                }
+                else if (failurePullingFiles)
+                {
+                    logger.LogError($"Received expected instrumentation exit code ({instrumentationExitCode}), but we hit errors pulling files from the device (see log for details.)");
+                    return ExitCode.DEVICE_FILE_COPY_FAILURE;
+                }
+                else
+                {
+                    return ExitCode.SUCCESS;
+                }
+            }
+            catch (Exception toLog)
+            {
+                logger.LogCritical(toLog, $"Failure to run test package: {toLog.Message}");
+            }
+
+            return ExitCode.GENERAL_FAILURE;
+        }
+
+        private (Dictionary<string, string> values, int exitCode) ParseInstrumentationOutputs(ILogger logger, string stdOut)
+        {
+            // If ADB.exe's output changes (which we control when we take updates in this repo), we'll need to fix this.
+            string resultPrefix = "INSTRUMENTATION_RESULT:";
+            string exitCodePrefix = "INSTRUMENTATION_CODE:";
+            int exitCode = -1;
+            var outputs = new Dictionary<string, string>();
+            string[] lines = stdOut.Split(Environment.NewLine);
+
+            foreach (string line in lines)
```

Comment:
Could we reduce the nesting by continuing when matching the given case rather that if/else/elsifing the flow?

Reply:
I see this is a refactored code form another location, so feel free to ignore my comment if you don't want to touch this.